orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/CreateServerGroupStage.groovy

```diff
@@ -58,4 +68,67 @@ class CreateServerGroupStage extends AbstractDeployStrategyStage {
 
     return tasks
   }
+
+  @Override
+  List<Stage> onFailureStages(@Nonnull Stage stage) {
+    def stageData = stage.mapTo(StageData)
+
+    if (!stageData.rollbackOnFailure) {
+      // rollback on failure is not enabled
+      return []
+    }
+
+    if (!stageData.getServerGroup()) {
+      // did not get far enough to create a new server group
```

Comment:
would it make sense to add a log statement here? Might help debugging issues like "why did my thing not get rolled back?"

Reply:
👍